File: app/src/main/java/mx/edu/upqroo/kristenandroid/adapters/source/NewsDataSourceFactory.java
package mx.edu.upqroo.kristenandroid.adapters.source;

import androidx.lifecycle.MutableLiveData;
import androidx.paging.DataSource;
import androidx.paging.PageKeyedDataSource;
import mx.edu.upqroo.kristenandroid.data.models.News;

public class NewsDataSourceFactory extends DataSource.Factory {

    private MutableLiveData<PageKeyedDataSource<Integer, News>> itemLiveDataSource;

    @Override
    public DataSource<Integer, News> create() {
        NewsDataSource itemDataSource = new NewsDataSource();

        itemLiveDataSource = new MutableLiveData<>();
        itemLiveDataSource.postValue(itemDataSource);

        return itemDataSource;
    }

    public MutableLiveData<PageKeyedDataSource<Integer, News>> getLiveDataSource() {
        return itemLiveDataSource;
    }
}

File: app/src/main/java/mx/edu/upqroo/kristenandroid/adapters/source/NewsDataSourceFactory.kt
package mx.edu.upqroo.kristenandroid.adapters.source

import androidx.lifecycle.MutableLiveData
import androidx.paging.DataSource
import androidx.paging.PageKeyedDataSource
import mx.edu.upqroo.kristenandroid.data.models.News

class NewsDataSourceFactory : DataSource.Factory<Int, News>() {

    var liveDataSource: MutableLiveData<PageKeyedDataSource<Int, News>>? = null
        private set

    override fun create(): DataSource<Int, News> {
        val itemDataSource = NewsDataSource()

        liveDataSource = MutableLiveData()
        liveDataSource!!.postValue(itemDataSource)

        return itemDataSource
    }
}